assembl/static2/js/app/components/cookies/cookiesSelectorContainer.jsx

```diff
@@ -0,0 +1,162 @@
+// @flow
+import React from 'react';
+import { compose, graphql } from 'react-apollo';
+import { connect } from 'react-redux';
+
+import CookiesSelector from './cookiesSelector';
+import { getCookieItem, setCookieItem } from '../../utils/globalFunctions';
+
+// graphql
+import acceptedCookiesQuery from '../../graphql/acceptedCookiesQuery.graphql';
+import updateAcceptedCookies from '../../graphql/mutations/updateAcceptedCookies.graphql';
+import withLoadingIndicator from '../common/withLoadingIndicator';
+
+import type { CookieObject } from './cookieToggle';
+import { COOKIE_TRANSLATION_KEYS } from '../../constants';
+
+type Props = {
+  updateAcceptedCookies: Function,
+  cookiesList: Array<string>
+};
+
+export type CookiesObject = {
+  other?: Array<CookieObject>,
+  essential?: Array<CookieObject>,
+  analytics?: Array<CookieObject>
+};
+
+type State = {
+  activeKey: ?string,
+  show: boolean,
+  cookies: ?CookiesObject
+};
+
+class CookiesSelectorContainer extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    const cookiesFromBrowser = getCookieItem('cookies_configuration');
+    const cookiesList = props.cookiesList ?
+      // if the user is logged in, we get the cookiesList from the query
+      props.cookiesList
+      // otherwise, we take it from the browser
+      : cookiesFromBrowser && cookiesFromBrowser.split(',');
+
+    const cookiesArray = cookiesList && cookiesList.map(
+      cookie => ({ ...this.getCookieObject(cookie), accepted: this.isCookieAccepted(cookie), cookieType: cookie })
+    );
+    const cookiesByCategory = cookiesArray && this.getCookiesObjectFromArray(cookiesArray);
+
+    this.state = {
+      activeKey: 'essential',
+      show: true,
+      cookies: cookiesByCategory
+    };
+  }
+
+  isCookieAccepted = (cookie: string) => {
+    if (cookie.startsWith('ACCEPT')) {
+      return true;
+    }
+    return false;
+  }
+
+  getCookieObject = (cookie: string) => {
+    if (cookie.includes('SESSION_ON_DISCUSSION')) {
+      return { category: 'other', name: COOKIE_TRANSLATION_KEYS.userSession, hasChanged: false };
+    }
+    if (cookie.includes('LOCALE_ON_DISCUSSION')) {
+      return { category: 'essential', name: COOKIE_TRANSLATION_KEYS.locale, hasChanged: false };
+    }
+    if (cookie.includes('TRACKING_ON_DISCUSSION')) {
+      return { category: 'analytics', name: COOKIE_TRANSLATION_KEYS.piwik, hasChanged: false };
+    }
+    return { category: 'other', name: cookie, hasChanged: false };
+  };
+
+  getCookiesObjectFromArray = (cookiesArray: Array<CookieObject>) => {
+    const cookiesObject = {};
+    cookiesArray.forEach(function(cookie) { // eslint-disable-line
+      const { category } = cookie;
+      if (cookiesObject[category]) {
+        cookiesObject[category] = [...cookiesObject[category], cookie];
+      } else {
+        cookiesObject[category] = [cookie];
+      }
+    });
+    return cookiesObject;
+  }
```

Comment:
it would be nice to have unit tests for those methods: `isCookieAccepted`, `getCookieObject` and `getCookiesObjectFromArray`. Take a look at https://github.com/assembl/assembl/blob/5216fd3746ff3be83634b98ab57d7782191e1a0d/assembl/static2/tests/unit/components/search/UserMessagesTagFilter.spec.jsx to see how to tests a component's methods

I feel like I could use such tests as documentation if I need to change something in this component.

Reply:
Yes, I planned to write tests for this component and CookiesSelector